lib/ui/window.dart

```diff
@@ -365,11 +407,95 @@ class Window {
   /// Message handlers must call the function given in the `callback` parameter.
   /// If the handler does not need to respond, the handler should pass `null` to
   /// the callback.
-  PlatformMessageCallback onPlatformMessage;
+  PlatformMessageCallback get onPlatformMessage => _onPlatformMessage;
+
+  set onPlatformMessage(PlatformMessageCallback callback) {
+    _onPlatformMessage = _zonedPlatformMessageCallback(callback);
+  }
+
+  PlatformMessageCallback _onPlatformMessage;
 
   /// Called by [_dispatchPlatformMessage].
   void _respondToPlatformMessage(int responseId, ByteData data)
       native "Window_respondToPlatformMessage";
+
+  /// Wraps the given [callback] in another callback that ensures that the
+  /// original callback is called in the zone it was registered in.
+  static VoidCallback _zonedVoidCallback(VoidCallback callback) {
```

Comment:
Another reason to do this is that the way you have it implemented, this will act weird:

```dart
// in zone A:
onFoo = () { };
// in zone B:
onFoo(); // now it's run in zone A, even though there's no obvious reason for it to do so
```

This might affect the test framework in subtle but confusing ways, since the test framework fiddles with these setters.